File: ParallaxEffect/app/src/main/java/com/example/parallaxeffect/ViewPageAdapter.java
package com.example.parallaxeffect;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import java.util.ArrayList;
import java.util.List;

@SuppressWarnings("deprecation")
public class ViewPageAdapter extends FragmentPagerAdapter {

    List<Fragment> fragmentList = new ArrayList<>();
    List<String> titlelist = new ArrayList<>();


    public ViewPageAdapter(@NonNull FragmentManager fm) {
        super(fm);
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        return fragmentList.get(position);
    }

    @Override
    public int getCount() {
        return fragmentList.size();
    }

    public void addFragment(Fragment fragment,String title)
    {
        fragmentList.add(fragment);
        titlelist.add(title);
    }


    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {

        return titlelist.get(position);
    }



}

File: Rahoul3pm/ParallaxeEffectinkotlin/app/src/main/java/com/example/parallaxeeffectinkotlin/ViewPageAdapter.kt
package com.example.parallaxeeffectinkotlin

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

class ViewPageAdapter(fragementmangaer: FragmentManager) : FragmentPagerAdapter(fragementmangaer) {

    var fragmelist: ArrayList<Fragment> = ArrayList()
    var titlelist: ArrayList<String> = ArrayList()
    override fun getCount(): Int {
       return fragmelist.size
    }

    override fun getItem(position: Int): Fragment {
       return fragmelist.get(position)
    }

    fun getfragment(fragment: Fragment,title:String)
    {
        fragmelist.add(fragment)
        titlelist.add(title)
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return titlelist.get(position)
    }
}